### ISSUE TITLE
Enable SqlCredentialFactory to load credentials from a classpath resource

### ISSUE THREAD

**Issue by meltsufin** [spring-cloud/spring-cloud-gcp#774]
Currently only files on the filesystem are supported because SqlCredentialFactory doesn't have access to the application context.
See [comment](https://github.com/spring-cloud/spring-cloud-gcp/issues/339#issuecomment-399532888).

**Comment by PTanwarCS** [comment]
Any updates please?

**Comment by meltsufin** [comment]
This has been depoirritized because, generally, it doesn't seem like a good practice to store credentials on the classpath.
What's your use-case?

**Comment by PTanwarCS** [comment]
Yeah it's not a good practice. Actually I was doing a POC where I needed configuration with minimal efforts.
